statusHandler/persister/persistentHandler.go

```diff
@@ -0,0 +1,200 @@
+package persister
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("statusHandler/persister")
+
+// PersistentStatusHandler is a status handler that will save metrics in storage
+type PersistentStatusHandler struct {
+	store                    storage.Storer
+	persistentMetrics        *sync.Map
+	marshalizer              marshal.Marshalizer
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
+	startSaveInStorage       bool
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
+) (*PersistentStatusHandler, error) {
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(uint64ByteSliceConverter) {
+		return nil, process.ErrNilUint64Converter
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
+	psh.uint64ByteSliceConverter = uint64ByteSliceConverter
+	psh.marshalizer = marshalizer
+	psh.persistentMetrics = &sync.Map{}
+	psh.initMap()
+
+	go func() {
+		time.Sleep(time.Second)
+
+		psh.startSaveInStorage = true
+	}()
+
+	return psh, nil
+}
+
+func (psh *PersistentStatusHandler) initMap() {
+	initUint := uint64(0)
+
+	psh.persistentMetrics.Store(core.MetricCountConsensus, initUint)
+	psh.persistentMetrics.Store(core.MetricCountConsensusAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountLeader, initUint)
+	psh.persistentMetrics.Store(core.MetricNumProcessedTxs, initUint)
+	psh.persistentMetrics.Store(core.MetricNumShardHeadersProcessed, initUint)
+	psh.persistentMetrics.Store(core.MetricNonce, initUint)
+}
+
+// SetStorage will set storage for persistent status handler
+func (psh *PersistentStatusHandler) SetStorage(store storage.Storer) error {
+	if check.IfNil(store) {
+		return process.ErrNilStorage
+	}
+
+	psh.store = store
+
+	return nil
+}
+
+func (psh *PersistentStatusHandler) saveMetricsInDb(nonce uint64) {
+	metricsMap := make(map[string]interface{})
+	psh.persistentMetrics.Range(func(key, value interface{}) bool {
+		keyString, ok := key.(string)
+		if !ok {
+			return false
+		}
+		metricsMap[keyString] = value
+		return true
+	})
+
+	statusMetricsBytes, err := psh.marshalizer.Marshal(&metricsMap)
```

Comment:
I think you can use metricsMap instead &metricsMap

Reply:
done